testing/test-utils/src/lib/utils/os-agnostic-paths.unit.test.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  MockInstance,
+  afterEach,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from 'vitest';
+import { osAgnosticPath } from './os-agnostic-paths';
+
+describe('osAgnosticPath', () => {
+  const cwdSpy: MockInstance<[], string> = vi.spyOn(process, 'cwd');
+
+  it('should forward nullish paths on Linux/macOS and Windows', () => {
+    expect(osAgnosticPath(undefined)).toBeUndefined();
+  });
+
+  describe('Unix-based systems (Linux/macOS)', () => {
+    const linuxmMacOSCWD = '/Users/jerry';
```

Comment:
This variable name is very hard to read, I would suggest renaming it.

```suggestion
    const unixCwd = '/Users/jerry';
```